src/VisualStudio/Core/Def/Implementation/ProjectSystem/RunningDocumentTableEventSink.cs

```diff
@@ -0,0 +1,243 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.ComponentModel.Composition;
+using Microsoft.CodeAnalysis.Editor.Shared.Utilities;
+using Microsoft.VisualStudio.Editor;
+using Microsoft.VisualStudio.Shell;
+using Microsoft.VisualStudio.Shell.Interop;
+using Microsoft.VisualStudio.Text;
+using Microsoft.VisualStudio.TextManager.Interop;
+
+namespace Microsoft.VisualStudio.LanguageServices.Implementation.ProjectSystem
+{
+    /// <summary>
+    /// Class to register with the RDT and forward RDT events.
+    /// Handles common conditions before sending out notifications.
+    /// </summary>
+    [Export(typeof(RunningDocumentTableEventSink))]
```

Comment:
I am fine with doing this through MEF, but you could make it just ordinary type which accept the types as constructor input.

Reply:
Sure, I can change it.  MEF made it easier for the Miscellaneous workspace impl, but harder for the open file tracker.